solidity-plugin/src/main/java/org/sonarsource/solidity/externalreport/AbstractExternalReportSensor.java

```diff
@@ -0,0 +1,131 @@
+      .name("Import of " + linterName() + " issues.");
+  }
+
+  @Override
+  public void execute(SensorContext sensorContext) {
+    boolean externalIssuesSupported = sensorContext.getSonarQubeVersion().isGreaterThanOrEqual(Version.create(7, 2));
+    String[] reportPaths = sensorContext.config().getStringArray(reportsPropertyName());
+    if (reportPaths.length == 0) {
+      return;
+    }
+
+    if (!externalIssuesSupported) {
+      LOG.error("Import of external issues requires SonarQube 7.2 or greater.");
+      return;
+    }
+
+    for (String reportPath : reportPaths) {
+      File report = getIOFile(sensorContext.fileSystem().baseDir(), reportPath);
+      importReport(sensorContext, report);
+    }
+  }
+
+  private void importReport(SensorContext context, File report) {
+    try {
+      String filename = null;
+      LOG.info(logPrefix() + "Importing {}", report.getPath());
+      for (String line : Files.readAllLines(report.toPath(), UTF_8)) {
+        if (!line.isEmpty()) {
+          ExternalIssue issue = parse(line);
+          if (issue != null) {
+            if (issue.message.equals("onlyfilename")) {
+              filename = issue.filename;
+            } else {
+              issue.filename = filename;
+              addLineIssue(context, issue);
+            }
+          }
+        }
+      }
+    } catch (IOException e) {
+      LOG.error(logPrefix() + "No issues information will be saved as the report file '{}' can't be read.",
+        report.getPath(), e);
+    }
+  }
+
+  static File getIOFile(File baseDir, String path) {
+    File file = new File(path);
+    if (!file.isAbsolute()) {
+      file = new File(baseDir, path);
+    }
+    return file;
+  }
+
+  InputFile getInputFile(SensorContext context, String filePath) {
+    FilePredicates predicates = context.fileSystem().predicates();
+    InputFile inputFile = context.fileSystem().inputFile(predicates.or(predicates.hasRelativePath(filePath), predicates.hasAbsolutePath(filePath)));
+    if (inputFile == null) {
+      LOG.warn(logPrefix() + "No input file found for {}. No {} issues will be imported on this file.", filePath, linterName());
+      return null;
+    }
+    return inputFile;
+  }
+
+  void addLineIssue(SensorContext context, ExternalIssue issue) {
+    InputFile inputFile = getInputFile(context, issue.filename);
+    if (inputFile != null) {
+      NewExternalIssue newExternalIssue = context.newExternalIssue();
+      NewIssueLocation primaryLocation = newExternalIssue.newLocation()
+        .message(issue.message)
+        .on(inputFile)
+        .at(inputFile.selectLine(issue.lineNumber));
+
+      newExternalIssue
+        .at(primaryLocation)
+        .forRule(RuleKey.of(issue.linter, issue.ruleKey))
+        .type(issue.type)
+        .severity(DEFAULT_SEVERITY)
+        .remediationEffortMinutes(DEFAULT_REMEDIATION_COST)
+        .save();
+    }
+  }
+
+  private static class ExternalRule {
```

Comment:
![Code Smell](https://sonarsource.github.io/sonar-github/code_smell.png 'Code Smell') Code Smell: Remove this unused private "ExternalRule" class. ([squid:S3985](https://sonarcloud.io/organizations/stylianos-agapiou-sonarsource-gi/rules#rule_key=squid%3AS3985))

[See it in SonarCloud](https://sonarcloud.io/project/issues?id=org.sonarsource.solidity%3Asonar-solidity&pullRequest=89&issues=AWUzG1tzWWqqPQVjGsm7&open=AWUzG1tzWWqqPQVjGsm7)

[ISSUE_KEY]: AWUzG1tzWWqqPQVjGsm7